control-plane/agents/core/src/node/registry.rs

```diff
@@ -57,8 +60,27 @@ impl Registry {
                     })
                 }
             }
-            Some(node) => Ok(node.read().await.node_state().clone()),
+            Some(node) => {
+                let mut copy_node_state = node.read().await.node_state().clone();
+                copy_node_state.drain_status = self.get_node_drain_status(node_id).await;
+                Ok(copy_node_state)
+            }
+        }
+    }
+
+    // deduce the drain state from the presence of drain labels and nexus instances
+    async fn get_node_drain_status(&self, node_id: &NodeId) -> DrainStatus {
```

Comment:
Probably should return a Result<> to handle the unwraps below

Reply:
Added error handling to this and to get_node_states() as a consequence.